.gen/greeter_agent.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+from base_agent import BaseAgent
+
+import grpc
+import json
+
+from pygnmi.client import gNMIclient
+
+from ndk import sdk_service_pb2
+from ndk import config_service_pb2
+
+class Greeter(BaseAgent):
+    def __init__(self, name):
+        super().__init__(name)
+
+    def __enter__(self):
+        super().__enter__()
+        self._subscribe()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        return super().__exit__(exc_type, exc_value, exc_traceback)
+    
+    def _subscribe(self):
+        op = sdk_service_pb2.NotificationRegisterRequest.AddSubscription
+        entry = config_service_pb2.ConfigSubscriptionRequest()
+        request = sdk_service_pb2.NotificationRegisterRequest(op=op, stream_id=self.stream_id, config=entry)
+
+        subscription_response = self.sdk_mgr_client.NotificationRegister(request=request, metadata=self.metadata)
+        logging.info(f"Status of configuration subscription response: {subscription_response.status}")
+    
+    def _get_name_from_configuration(self, obj) -> str:
+        data = json.loads(obj.config.data.json)
+        name = data["name"]["value"] if "name" in data else ""
+        return name
+    
+    def _set_uptime(self):
+        uptime = "8:00"
+        telemetry_data = {"uptime": uptime}
+        self._update_telemetry(f".{self.name}", telemetry_data)
+
+    def _handle_notification(self, obj):        
+        logging.info("Process notification")
+        if obj.config.key.js_path == f".{self.name}":
+            name = self._get_name_from_configuration(obj)
+            if name == "":
+               logging.info("greeter/name not configured") 
```

Comment:
Can greeters have names that are just spaces? (" "). `_get_name_from_configuration()` returns an empty string when no name are given in the config, should we be checking that the name contains characters?

Reply:
> Can greeters have names that are just spaces? (" "). `_get_name_from_configuration()` returns an empty string when no name are given in the config, should we be checking that the name contains characters?

Spaces are not allowed by SRL CLI so checking for empty string is enough.

Reply:
This would ensure that " " also logs as invalid
```suggestion
            if name.strip() == "":
               logging.info("greeter/name not configured") 
```

Reply:
> This would ensure that " " also logs as invalid

Since it is not allowed to set `name` to an empty string with the SRL CLI command I will keep it like this. The goal is to log the name entered by the user as is and the uptime not to validate the data.